src/algebraic_topology/simplex_category.lean

```diff
@@ -0,0 +1,293 @@
+        (le_trans h_2 (nat.succ_le_of_lt h_1)))) },
+  swap 3,
+  { simp only [subtype.mk_lt_mk, not_lt] at h_1,
+    exact false.elim
+    (lt_irrefl j (lt_of_lt_of_le (nat.pred_lt_pred (nat.succ_ne_zero j) h_2) h_1)) },
+  -- Deal with the rest automatically.
+  all_goals { simp at *; linarith, },
+end
+
+end generators
+
+section skeleton
+
+lemma skeletal : skeletal simplex_category :=
+λ X Y I,
+begin
+  rcases I with ⟨I⟩,
+  suffices : fintype.card (fin (X+1)) = fintype.card (fin (Y+1)),
+  { simpa, },
+  { apply fintype.card_congr,
+    refine ⟨I.hom, I.inv, _, _⟩,
+    intro x, exact congr_fun (congr_arg (preorder_hom.to_fun) I.hom_inv_id : _) x,
+    intro y, exact congr_fun (congr_arg (preorder_hom.to_fun) I.inv_hom_id : _) y, }
+end
+
+/-- The functor that exhibits `simplex_category` as skeleton
+of `NonemptyFinLinOrd` -/
+def skeletal_functor :
+  simplex_category ⥤ NonemptyFinLinOrd.{u} :=
+{ obj := λ n, NonemptyFinLinOrd.of $ ulift (fin (n+1)),
+  map := λ m n f, ⟨λ i, ⟨f i.down⟩, λ ⟨i⟩ ⟨j⟩ h, show f i ≤ f j, from f.monotone h⟩, }
+
+namespace skeletal_functor
+
+instance : full skeletal_functor.{u} :=
+{ preimage := λ m n f, ⟨λ i, (f ⟨i⟩).down, λ i j h, f.monotone h⟩,
+  witness' := by { intros m n f, dsimp at *, ext1 ⟨i⟩, ext1, refl } }
+
+instance : faithful skeletal_functor.{u} :=
+{ map_injective' := λ m n f g h,
+  begin
+    ext1 i, apply equiv.ulift.symm.injective,
+    show skeletal_functor.map f ⟨i⟩ = skeletal_functor.map g ⟨i⟩,
+    rw h,
+  end }
+
+instance : ess_surj skeletal_functor.{u} :=
+{ mem_ess_image := λ X, ⟨(fintype.card X - 1 : ℕ), ⟨begin
+    have aux : fintype.card X = fintype.card X - 1 + 1,
+    { exact (nat.succ_pred_eq_of_pos $ fintype.card_pos_iff.mpr ⟨⊥⟩).symm, },
+    let f := mono_equiv_of_fin X aux,
+    have hf := (finset.univ.order_emb_of_fin aux).strict_mono,
+    refine
+    { hom := ⟨λ i, f i.down, _⟩,
+      inv := ⟨λ i, ⟨f.symm i⟩, _⟩,
+      hom_inv_id' := _,
+      inv_hom_id' := _ },
+    { rintro ⟨i⟩ ⟨j⟩ h, show f i ≤ f j, exact hf.monotone h, },
+    { intros i j h, show f.symm i ≤ f.symm j, rw ← hf.le_iff_le,
+      show f (f.symm i) ≤ f (f.symm j), simpa only [order_iso.apply_symm_apply], },
+    { ext1 ⟨i⟩, ext1, exact f.symm_apply_apply i },
+    { ext1 i, exact f.apply_symm_apply i },
+  end⟩⟩,}
+
+noncomputable instance is_equivalence : is_equivalence skeletal_functor.{u} :=
+equivalence.equivalence_of_fully_faithfully_ess_surj skeletal_functor
+
+end skeletal_functor
+
+/-- The equivalence that exhibits `simplex_category` as skeleton
+of `NonemptyFinLinOrd` -/
+noncomputable def skeletal_equivalence :
+  simplex_category ≌ NonemptyFinLinOrd.{u} :=
+functor.as_equivalence skeletal_functor.{u}
+
+end skeleton
+
+/--
+`simplex_category` is a skeleton of `NonemptyFinLinOrd`.
+-/
+noncomputable
+def is_skeleton_of : is_skeleton_of NonemptyFinLinOrd.{u} simplex_category skeletal_functor.{u} :=
+{ skel := skeletal,
+  eqv := is_equivalence.of_equivalence skeletal_equivalence }
```

Comment:
```suggestion
  eqv := skeletal_functor.is_equivalence }
```
very minor, but you show the functor's an equivalence, then make the equivalence of categories, then convert back, so might as well just go directly